tools/protobuf/IceRpc.ProtocGen/ServiceGenerator.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) ZeroC, Inc.
+
+using Google.Protobuf.Reflection;
+using IceRpc.CaseConverter.Internal;
+
+namespace IceRpc.ProtocGen;
+
+internal class ServiceGenerator
+{
+    public static string GenerateInterface(ServiceDescriptor service)
+    {
+        string methods = "";
+        foreach (MethodDescriptor method in service.Methods)
+        {
+            string inputType = method.InputType.GetFullyQualifiedType();
+            string returnType = method.OutputType.GetFullyQualifiedType();
+            string methodName = $"{method.Name.ToPascalCase()}Async";
+
+            // Add an abstract method to the interface for each service method.
+            methods += $@"
+    global::System.Threading.Tasks.ValueTask<{returnType}> {methodName}(
+        {inputType} message,
+        IceRpc.Features.IFeatureCollection features,
+        global::System.Threading.CancellationToken cancellationToken);";
+
+            // Add a static method for each service method, the implementation calls the
+            // abstract method and creates the outgoing response.
+            methods += $@"
+    [ProtobufOperation(""{method.Name}"")]
+    [global::System.ComponentModel.EditorBrowsable(global::System.ComponentModel.EditorBrowsableState.Never)]
+    protected static async global::System.Threading.Tasks.ValueTask<IceRpc.OutgoingResponse> {methodName}(
```

Comment:
I think it would be cleaner to add a prefix these methods like we do for Slice, e.g. ProtobufDXxx.

Reply:
It would be even nicer to generate this code from the source generator, but probably more difficult?

Reply:
We can try in a follow-up PR, the main difficulty is where to place these static methods. The source generator cannot put them in the service interface, because partials have to reside in the same assembly and we cannot assume that the interface and the service are always in the same assembly.